### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by andreas-bok-sociomantic**
Adds documentation for v5.4.0 API and previously omitted functions

Minor formatting fixes

### CODE REVIEW

File: docs/api/connection.rst [hunk 1]
```diff
@@ -204,6 +204,21 @@ Functions
...
+
+   :param options: The options object to modify
+   :param owner: The owner of the socket connection
+
+.. c:function:: drizzle_socket_owner drizzle_options_get_socket_owner(drizzle_options_st *options)
+
+   Gets the owner of the socket connection
+
+   :param options: The options object to get the value from
+   :returns: The owner of the socket
```
Comment (gautam-kotian-sociomantic) [inline docs/api/connection.rst]: You seem to always say "owner of the socket connection", so it seems odd that the word `connection` is missing here.

File: docs/api/constants.rst [hunk 2]
```diff
@@ -730,6 +730,17 @@ Connection
       :py:const:`DRIZZLE_CAPABILITIES_PROTOCOL_41`,
       :py:const:`DRIZZLE_CAPABILITIES_SECURE_CONNECTION`
 
+.. c:type:: drizzle_ssl_state_t
```
Comment (gautam-kotian-sociomantic) [inline docs/api/constants.rst:733]: The commit message doesn't mention anything about adding documentation for the enum. Do you want to add that, or do you want to roll these changes into a separate commit?

File: docs/api/connection.rst (hunk 1 above)
Comment (andreas-bok-sociomantic) [reply docs/api/connection.rst]: Will change.

File: docs/api/constants.rst (hunk 2 above)
Comment (andreas-bok-sociomantic) [reply docs/api/constants.rst:733]: Yeah. Would make sense with a separate commit.

**Comment by andreas-bok-sociomantic** [comment]
Updated based on comments from @gautam-kotian-sociomantic 
```diff
diff --git a/docs/api/connection.rst b/docs/api/connection.rst
index 4fd05c7..74e6787 100644
--- a/docs/api/connection.rst
+++ b/docs/api/connection.rst
@@ -217,7 +217,7 @@ Functions
    Gets the owner of the socket connection
 
    :param options: The options object to get the value from
-   :returns: The owner of the socket
+   :returns: The owner of the socket connection
 ```

**Comment by andreas-bok-sociomantic** [comment]
More comments?

**Comment by gautam-kotian-sociomantic** [comment]
LGTM.

**Comment by andreas-bok-sociomantic** [comment]
Allright, feel free to merge

**Pull request merged**